diff --git a/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt b/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt
index 13a3c24..31b39fb 100644
--- a/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt
+++ b/node_modules/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt
@@ -63,7 +63,9 @@ class AudioModule : Module() {
       AudioManager.AUDIOFOCUS_LOSS_TRANSIENT -> {
         focusAcquired = false
         players.values.forEach {
-          it.player.pause()
+          runOnMain {
+            it.player.pause()
+          }
         }
       }
 
@@ -71,7 +73,9 @@ class AudioModule : Module() {
         focusAcquired = false
         if (interruptionMode == InterruptionMode.DUCK_OTHERS) {
           players.values.forEach {
-            it.player.volume /= 2f
+            runOnMain {
+              it.player.volume /= 2f
+            }
           }
         }
       }
@@ -147,7 +151,7 @@ class AudioModule : Module() {
     AsyncFunction("setIsAudioActiveAsync") { enabled: Boolean ->
       audioEnabled = enabled
       if (!enabled) {
-        appContext.mainQueue.launch {
+        runOnMain {
           players.values.forEach {
             if (it.player.isPlaying) {
               it.player.pause()
